pkgs/tools/package-management/nix/default.nix

```diff
@@ -197,22 +197,20 @@ in lib.makeExtensible (self: ({
     # allocation function Clang uses with this setting actually works
     # all the way back to 10.6.
     stdenv = overrideSDK stdenv { darwinMinVersion = "10.13"; };
-  })).overrideAttrs (o: {
-    meta.knownVulnerabilities = [
-      "Nix >= 2.24.0 and master have a vulnerability. Please downgrade from nix_2_24 to nix_2_23"
-    ];
   });
 
-  git = ((common rec {
+  git = (common rec {
     version = "2.25.0";
-    suffix = "pre20240807_${lib.substring 0 8 src.rev}";
+    suffix = "pre20240910_${lib.substring 0 8 src.rev}";
     src = fetchFromGitHub {
       owner = "NixOS";
       repo = "nix";
-      rev = "cfe66dbec325d5dcb601b642bd9c149ae1353147";
-      hash = "sha256-1hqjl4br3MRK1pkzDrhBSxKUhdfQ/P4b5KbLfGua64g=";
+      rev = "9c0e96884355aab65b1879feeb81c5fea1719ab0";
+      hash = "sha256-1d0YwZPZ7Ygr22Dw1rBhgPDSlf24dhowKBf6QeyhSKg=";
     };
     self_attribute_name = "git";
+    # TODO: upstream this
```

Comment:
```suggestion
    # See https://github.com/NixOS/nix/pull/11472
```

Reply:
Another instance of https://github.com/NixOS/nix/pull/11138, I guess. I don’t think we ever figured out the reason for the platform divergence.

Reply:
> I don’t think we ever figured out the reason for the platform divergence.

That problematic PR actually passed macOS build CI: https://github.com/NixOS/nix/actions/runs/10567515856/job/29276631524

Reply:
Yeah, because they still only run CI for `aarch64-darwin` and the `x86_64-darwin` build keeps breaking as a result. It’s probably an SDK version difference, so 25.05 will fix it. Though relying on implicit imports is gross anyway.

Reply:
Fixed.